tools/regconvert.py

```diff
@@ -11,13 +12,105 @@ def get_reg_macro(offset):
     reg = (offset - DATA_OFFSET) // 2
     group = reg // GROUP_SIZE
     reg_in_group = reg % GROUP_SIZE
-    return "%cREG(%d)\n" % (REGISTER_NAMES[group], reg_in_group)
+    return "%cREG(%d)" % (REGISTER_NAMES[group], reg_in_group)
+
+def parse_number(string, force_hex):
+    if string.lower().startswith("0x") or force_hex:
+        return int(string, 16)
+    return int(string, 10)
+
+def index_to_offset(index):
+    return index * 2 + DATA_OFFSET
+
+def read_file(filename):
+    parsed_contents = ""
+
+    src_file = open(filename)
+    file_contents = src_file.read()
+    src_file.close()
+
+    pattern = re.compile("gGameInfo->data\[((0[xX])?[0-9a-fA-F]+)\]")
+
+    match = pattern.search(file_contents)
+    while match:
+        index = parse_number(match.group(1), False)
+        offset = index_to_offset(index)
+
+        start, end = match.span()
+        parsed_contents += file_contents[:start]
+        parsed_contents += get_reg_macro(offset)
+
+        file_contents = file_contents[end:]
+        match = pattern.search(file_contents)
+    
+    parsed_contents += file_contents
+    return parsed_contents
+
+def write_file(filename, contents):
+    dst_file = open(filename, "w")
```

Comment:
I think with file stuff it's cleaner to wrap the file opening in the with block so it gets automatically closed at the end of the block:

```
with open(filename, "w") as f:
    f.write(contents)
```